andela/learning-map-front
# #150637167 Fix token expiration issue

#### What does this PR do?
Implement middleware for token expiration validation

#### Description of Task to be completed?
This PR will complete the following task:
- Create token validation middleware
- Apply middleware in store to verify token expiration before making any request
- Redirect user to the login page when token expires

#### How should this be manually tested?
- Clone the project
- Run yarn install
- Run yarn start
- Log in with an Andela email address
- While on the `Dashboard`, replace the token in the cookie with this expired token
`eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJVc2VySW5mbyI6eyJpZCI6Ii1LY1lCSFJEWGd0ZHUtUHlpcEM5IiwiZW1haWwiOiJhYmR1bG11bWVlbi5vbGFzb2RlQGFuZGVsYS5jb20iLCJmaXJzdF9uYW1lIjoiQWJkdWxtdW1lZW4iLCJsYXN0X25hbWUiOiJPbGFzb2RlIiwibmFtZSI6IkFiZHVsbXVtZWVuIE9sYXNvZGUiLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tLy0tZTFVN3JNTElXRS9BQUFBQUFBQUFBSS9BQUFBQUFBQUFDby84S0F2ME91TDBTWS9waG90by5qcGc_c3o9NTAiLCJyb2xlcyI6eyJBbmRlbGFuIjoiLUtpaWhmWm9zZVFlcUM2YldUYXUiLCJGZWxsb3ciOiItS1hHeTFFQjFvaW1qUWdGaW02QyJ9fSwiZXhwIjoxNTA2NjE0MH0.nyabG0SeY_bTYSnposTxCEIL5GDQsO9fBZV-v8ptOQo`
- Refresh the page to confirm that you are logged out

#### Any background context you want to provide?
User is still logged in after token expires but resources from the backend are not available.

#### What are the relevant pivotal tracker stories?
[#150637167](https://www.pivotaltracker.com/story/show/150637167)

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
